src/app/filters/filter-fields.component.ts

```diff
@@ -27,11 +30,16 @@ export class FilterFieldsComponent implements OnInit {
   @Input() config: FilterConfig;
 
   @Output('onAdd') onAdd = new EventEmitter();
+  @Output('onSelectType') onSelecttype = new EventEmitter();
+
+  @ViewChild('searchField') searchField: any;
+  @ViewChild('listData') listData: any;
```

Comment:
Adding listData here only supports one filter field type -- this means there can only ever be one type-ahead field. The filter config is intended to multiple filter field types, each with their own unique properties.